babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1583,3 +1583,49 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
             SqlLiteral.createBoolean(withTies, pos));
     }
 }
+
+SqlNode SqlHexCharStringLiteral() :
+{
+    final NlsString p;
+    final String formatString;
+    final String charSet;
+    CharacterSet charSetEnum = null;
+    final HexCharLiteralFormat format;
+}
+{
+    (
+        <PREFIXED_HEX_STRING_LITERAL>
+        {
+            charSet = SqlParserUtil
```

Comment:
This looks like a lot of Java logic in a parser file. Can we move it down into the constructor of SqlHexCharString, which we pass the raw charSet?

Reply:
This suggestion looks reasonable, but since we would need to call the superclass constructor at the first line of the constructor of SqlHexCharString, it seems like we still need a small part of java logic here. What I can do is to move most of the parsing logic into the constructor of SqlHexCharString, but there will probably some left.

Reply:
Refactoring done in https://github.com/googleinterns/calcite/pull/155/commits/eb9c2fd414f98e835b3ef2d007f0093d7df4e3d9 to address this issue